### PR TITLE
Fixed frequent width changes of and slider and time indicator in touchbar when sliding

### PR DESCRIPTION
**Pull request by csujedihy**
When a user moves the slider to seek a new position for the video, the width of the slider and the time indicator on the right change frequently due to fitting the content (ex. 1:03 -> 1:23:09).

I added a width constraint to the NSTextField when loading video file and used a dummy NSTextField to estimate the width.

### CODE REVIEW

**Comment by lhc70000** [comment]
Switched to `develop` branch for developing. Please change the target branch.

**Comment by csujedihy** [comment]
Changed to `develop`. Plz review.

File: Podfile.lock [hunk 1]
```diff
@@ -9,4 +9,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: 1c4e3202e88a4154f713ccae52801c835a85e231
 
-COCOAPODS: 1.2.0.beta.1
+COCOAPODS: 1.1.1
```
Comment (lhc70000) [inline Podfile.lock]: Please discard all changes to `Podfile.lock`

File: iina.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1166,6 +1169,7 @@
 				"OTHER_SWIFT_FLAGS[arch=*]" = "-DRELEASE";
 				PRODUCT_BUNDLE_IDENTIFIER = com.colliderli.iina;
 				PRODUCT_NAME = IINA;
+				PROVISIONING_PROFILE = "";
 				SKIP_INSTALL = NO;
 				SWIFT_OBJC_BRIDGING_HEADER = "iina/iina-Bridging-Header.h";
 				SWIFT_OPTIMIZATION_LEVEL = "-Owholemodule";
```
Comment (lhc70000) [inline iina.xcodeproj/project.pbxproj]: Please discard all changes to `iina.xcodeproj/project.pbxproj`

Comment (csujedihy) [inline iina.xcodeproj/project.pbxproj]: I will resubmit PR.

File: iina/MainWindowController.swift [hunk 3]
```diff
@@ -462,6 +466,29 @@ class MainWindowController: NSWindowController, NSWindowDelegate {
...
+        posLabel.setNeedsDisplay()
+      } else {
+        let posConstraint = NSLayoutConstraint(item: posLabel, attribute: .width, relatedBy: .equal, toItem: nil, attribute: .notAnAttribute, multiplier: 1.0, constant: widthConstant + pad)
+        posLabel.addConstraint(posConstraint)
+        touchBarPosLabelWidthLayout = posConstraint
+      }
+    }
+
+  }
+
```
Comment (lhc70000) [inline iina/MainWindowController.swift]: Please move this function to the end of `NSTouchBarDelegate` extension, i.e. line 1346.

Also add a system version check in `PlayerCore` (Xcode will show a warning and do this for you).

File: iina/MainWindowController.swift [hunk 4]
```diff
@@ -1177,6 +1205,7 @@ fileprivate let touchBarItemBinding: [NSTouchBarItemIdentifier: (String, Int, St
 
 @available(OSX 10.12.2, *)
 extension MainWindowController: NSTouchBarDelegate {
+
   
   override func makeTouchBar() -> NSTouchBar? {
     let touchBar = NSTouchBar()
```
Comment (lhc70000) [inline iina/MainWindowController.swift]: Please add a `label.alignment = .center` to the text label (when create it, in `case NSTouchBarItemIdentifier.time:`) to make it looks better.

**Comment by lhc70000** [comment]
It works well. Thanks a lot! 😃 

There only exist some minor issue, as commented above.

**Comment by csujedihy** [comment]
I think I gotta fork the project again and applies such changes. 

For this one,
> add a system version check in PlayerCore (Xcode will show a warning and do this for you).

I don't quite understand. I just added one line of code in PlayerCore.swift.

**Comment by csujedihy** [comment]
I got it. The system version check should be in `setupTouchBarUI()`

**Comment by lhc70000** [comment]
If you move `setupTouchBarUI()` to `NSTouchBarDelegate` extension, which is decorated by `@available(OSX 10.12.2, *)`, when you call it in `PlayerCore`, Xcode will show a warn and auto fix it like

```
if #available(OSX 10.12.2, *) {
  setupTouchBarUI()
}
```

I think i's better to check it in `PlayerCore` before calling the whole function.

**Comment by csujedihy** [comment]
Yes. That makes sense.

**Pull request closed without merging**